### PR TITLE
feat: Allow gas fee scaler's to be settable via env variables

### PR DESCRIPTION
**Pull request by nicholaspai**
So we can experiment with relayer gas pricing

### CODE REVIEW

File: src/utils/TransactionUtils.ts [hunk 1]
```diff
@@ -20,7 +21,11 @@ export async function runTransaction(
   nonce: number | null = null
 ): Promise<TransactionResponse> {
   try {
-    const gas = await getGasPrice(contract.provider);
+    const gas = await getGasPrice(
+      contract.provider,
+      process.env.PRIORITY_FEE_SCALER && Number(process.env.PRIORITY_FEE_SCALER),
+      process.env.MAX_FEE_PER_GAS_SCALER && Number(process.env.MAX_FEE_PER_GAS_SCALER)
+    );
```
Comment (mrice32) [inline src/utils/TransactionUtils.ts:28]: Thoughts on something like this where we have a chain-specific one and then an omnichain one to fall back to? That way we can tune it for each chain if we like. Something like:

```suggestion
    const priorityFeeScaler = Number(process.env[`PRIORITY_FEE_SCALER_${chainId}`] || process.env.PRIORITY_FEE_SCALER) || undefined;
    const maxFeePerGasScaler = Number(process.env[`MAX_FEE_PER_GAS_SCALER_${chainId}`] || process.env.MAX_FEE_PER_GAS_SCALER) || undefined;

    const gas = await getGasPrice(
      contract.provider,
      priorityFeeScaler,
      maxFeePerGasScaler
    );
```

**Review by mrice32 — COMMENTED** [review]
Just one minor comment!

File: src/utils/TransactionUtils.ts (hunk 1 above)
Comment (pxrl) [reply src/utils/TransactionUtils.ts:28]: Yeah it'd be useful for this to be per-chain configurable.

Comment (nicholaspai) [reply src/utils/TransactionUtils.ts]: Great idea!

File: src/utils/TransactionUtils.ts [hunk 2]
```diff
@@ -20,7 +21,12 @@ export async function runTransaction(
   nonce: number | null = null
 ): Promise<TransactionResponse> {
   try {
-    const gas = await getGasPrice(contract.provider);
+    const priorityFeeScaler =
+      Number(process.env[`PRIORITY_FEE_SCALER_${chainId}`] || process.env.PRIORITY_FEE_SCALER) || undefined;
```
Comment (pxrl) [inline src/utils/TransactionUtils.ts:25]: I think you can fish the chainId out from contract.provider.getNetwork().chainId (or something similar).

Comment (nicholaspai) [reply src/utils/TransactionUtils.ts:26]: yeah this adds an RPC request but `runTransaction` is once per run so i think this is OK, WDYT?

Comment (nicholaspai) [reply src/utils/TransactionUtils.ts:26]: Alternatively we could try to load in chain ID from calling code where possible (sometimes its already loaded) and then add an RPC request where its not

Comment (pxrl) [reply src/utils/TransactionUtils.ts:26]: We use StaticJsonRpcProvider so the lookup just returns ethers' local copy of the chainId - so there should be no cost to this :+1:

**Review by pxrl — APPROVED** [review]

**Pull request merged**